clients/java/src/test/java/io/camunda/zeebe/client/usertask/AssignUserTaskTest.java

```diff
@@ -0,0 +1,96 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.usertask;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.camunda.zeebe.client.api.command.ProblemException;
+import io.camunda.zeebe.client.protocol.rest.ProblemDetail;
+import io.camunda.zeebe.client.protocol.rest.UserTaskAssignmentRequest;
+import io.camunda.zeebe.client.util.ClientRestTest;
+import io.camunda.zeebe.client.util.RestGatewayService;
+import org.junit.jupiter.api.Test;
+
+public final class AssignUserTaskTest extends ClientRestTest {
+
+  @Test
+  void shouldAssignUserTask() {
+    // when
+    client.newUserTaskAssignCommand(123L).assignee("foo").send().join();
+
+    // then
+    final UserTaskAssignmentRequest request =
+        gatewayService.getLastRequest(UserTaskAssignmentRequest.class);
+    assertThat(request.getAction()).isNull();
+    assertThat(request.getAllowOverride()).isNull();
+    assertThat(request.getAssignee()).isEqualTo("foo");
+  }
+
+  @Test
+  void shouldAssignUserTaskWithAction() {
+    // when
+    client.newUserTaskAssignCommand(123L).action("foo").send().join();
+
+    // then
+    final UserTaskAssignmentRequest request =
+        gatewayService.getLastRequest(UserTaskAssignmentRequest.class);
+    assertThat(request.getAction()).isEqualTo("foo");
+    assertThat(request.getAllowOverride()).isNull();
+    assertThat(request.getAssignee()).isNull();
+  }
+
+  @Test
+  void shouldAssignUserTaskWithAllowOverride() {
+    // when
+
+    client.newUserTaskAssignCommand(123L).allowOverride(true).send().join();
+
+    // then
+    final UserTaskAssignmentRequest request =
+        gatewayService.getLastRequest(UserTaskAssignmentRequest.class);
+    assertThat(request.getAction()).isNull();
+    assertThat(request.getAllowOverride()).isTrue();
+    assertThat(request.getAssignee()).isNull();
+  }
+
+  @Test
+  void shouldAssignUserTaskWithAllowOverrideDisabled() {
+    // when
+
+    client.newUserTaskAssignCommand(123L).allowOverride(false).send().join();
+
+    // then
+    final UserTaskAssignmentRequest request =
+        gatewayService.getLastRequest(UserTaskAssignmentRequest.class);
+    assertThat(request.getAction()).isNull();
+    assertThat(request.getAllowOverride()).isFalse();
+    assertThat(request.getAssignee()).isNull();
+  }
+
+  @Test
+  void shouldRaiseExceptionOnError() {
+    // given
+    gatewayService.errorOnRequest(
+        String.format(RestGatewayService.URL_USER_TASK_ASSIGNMENT, 123L),
```

Comment:
🔧 If you happen to move the template strings into a separate class, you could provide a method to get the URL. It can be implemented with this same call to `String.format`, but provide more clarity to the reader. For example, this could allow method chaining to clarify intent (although perhaps a bit overengineered):

```java
gatewayService.errorOnRequest(
    GatewayServicePaths.resolveUrl(resource: USER_TASK).key(123L).assignment()
    () -> new ProblemDetail().title("Not Found").status(404));
```

💭 This also makes me think that it may be useful to allow passing a specific request method to the `errorOnRequest`, so tests have more control over the specific request that's mocked.

Reply:
I played around with that for a bit but I think we're really over-engineering this for what we currently need. I settled for a "middle ground", making the URL templates internal and providing respective static getters that clarify what arguments are required to build a proper URL for each endpoint.

Let me know what you think.

Reply:
I like it! 👍